File: ION/packages/Nfinium/XFINIUM.PDF 5.4.0 Mobile Professional Edition Commercial Version/Samples/Common/DocumentSplit.cs
using System;
using System.IO;
using Xfinium.Pdf;
using Xfinium.Pdf.Graphics;
using Xfinium.Pdf.Destinations;
using Xfinium.Pdf.Actions;
using Xfinium.Pdf.Core;

namespace Xfinium.Pdf.Samples
{
    /// <summary>
    /// DocumentSplit sample.
    /// </summary>
    public class DocumentSplit
    {
        /// <summary>
        /// Main method for running the sample.
        /// </summary>
        public static SampleOutputInfo[] Run(Stream input)
        {
            // The input file is split by extracting pages from source file and inserting them in new empty PDF documents.
            PdfFile file = new PdfFile(input);
            SampleOutputInfo[] output = new SampleOutputInfo[file.PageCount];

            for (int i = 0; i < file.PageCount; i++)
            {
                PdfFixedDocument document = new PdfFixedDocument();
                PdfPage page = file.ExtractPage(i);
                document.Pages.Add(page);

                output[i] = new SampleOutputInfo(document, string.Format("xfinium.pdf.sample.documentsplit.{0}.pdf", i + 1));
            }

            return output;
        }
    }
}

File: ION/packages/Nfinium/XFINIUM.PDF 5.4.0 Mobile Professional Edition Commercial Version/Samples/Common/DocumentSplit.vb
Imports System.IO
Imports Xfinium.Pdf
Imports Xfinium.Pdf.Graphics
Imports Xfinium.Pdf.Destinations
Imports Xfinium.Pdf.Actions
Imports Xfinium.Pdf.Core

Namespace Xfinium.Pdf.Samples
	''' <summary>
	''' DocumentSplit sample.
	''' </summary>
	Public Class DocumentSplit
		''' <summary>
		''' Main method for running the sample.
		''' </summary>
		Public Shared Function Run(input As Stream) As SampleOutputInfo()
			' The input file is split by extracting pages from source file and inserting them in new empty PDF documents.
			Dim file As New PdfFile(input)
			Dim output As SampleOutputInfo() = New SampleOutputInfo(file.PageCount - 1) {}

			For i As Integer = 0 To file.PageCount - 1
				Dim document As New PdfFixedDocument()
				Dim page As PdfPage = file.ExtractPage(i)
				document.Pages.Add(page)

				output(i) = New SampleOutputInfo(document, String.Format("xfinium.pdf.sample.documentsplit.{0}.pdf", i + 1))
			Next

			Return output
		End Function
	End Class
End Namespace
